src/p2p/net_node.inl

```diff
@@ -1400,19 +1539,30 @@ namespace nodetool
     std::list<peerlist_entry> peerlist_ = peerlist;
     if(!fix_time_delta(peerlist_, local_time, delta))
       return false;
+
+    const epee::net_utils::zone zone = context.m_remote_address.get_zone();
+    for(const auto& peer : peerlist_)
+    {
+      if(peer.adr.get_zone() != zone)
+      {
+        MWARNING("Connection from " << context.m_remote_address.str() << " sent peerlist from another zone, dropping");
```

Comment:
When a log is associated with a peer, I try (when I remember) to use MWARNING(context << "blah blah"); as this makes logs easier to read and grep.